packages/react/src/jsx/ReactJSXElement.js

```diff
@@ -239,10 +239,8 @@ export function jsx(type, config, maybeKey) {
       hasOwnProperty.call(config, propName) &&
       // Skip over reserved prop names
       propName !== 'key' &&
-      // TODO: These will no longer be reserved in the next major
-      propName !== 'ref' &&
-      propName !== '__self' &&
-      propName !== '__source'
```

Comment:
These really shouldn't have been filtered out from the start since they're passed as separate arguments in this transform.